htlcswitch/switch.go

```diff
@@ -2367,6 +2387,16 @@ func (s *Switch) addLiveLink(link ChannelLink) {
 			s.baseIndex[alias] = linkScid
 		}
 	} else if link.negotiatedAliasFeature() {
+		// First, we flush any alias mappings for this link's scid
+		// before we populate the map again, in order to get rid of old
+		// values that no longer exist.
+		for alias, real := range s.aliasToReal {
```

Comment:
Following up on https://github.com/lightningnetwork/lnd/pull/8509/commits/c6576187e9f445df9f63d1837cd97020d2837f2d#r1580115154: I think @Crypt-iQ does mean that we also need to the same for the `s.baseIndex` map. Otherwise we will keep old alias mapping around.

Reply:
So basically:

```
for alias, real := range s.baseIndex {
    if real == linkScid {
        delete(s.baseIndex, alias)
    }
}

```

Reply:
oh nice catch, added this part too :+1: